server/services/build-service.js

```diff
@@ -24,6 +25,19 @@ module.exports = ({ strapi }) => ({
 			if (settings.body) {
 				requestConfig.data = settings.body;
 			}
+
+			if (settings.secret) {
+				const { hash, headerKey, secretKey } = settings.secret;
+				const normalizedHash = hash.toLowerCase();
+				requestConfig.headers = {
+					...(requestConfig.headers || {}),
```

Comment:
I dont see a reason to do use a spread here, would rather prefer a shorted check.

```js
requestConfig.headers = requestConfig.headers || {}
```
Maybe we should instead add an empty headers obj same as we do for data? This will remove any need for a check, i dont believe it should cause any issues.